### ISSUE TITLE
improve node running guide for setting rpc endpoints

### ISSUE THREAD

**Issue by d1onys1us** [taikoxyz/taiko-mono#13986]
![Image](https://github.com/taikoxyz/taiko-mono/assets/13951458/51a50dee-d755-4ddf-8b3d-f035077b9d47)

this is too vague for the user. we can update it with something like:  Make an Alchemy account -> Create new app -> Choose Sepolia -> View Key -> Place http... and wss... address in .env file -> Save

**Comment by Eleleth77** [comment]
**Got it working. Suggestion:**

Finally, set the following environment variables:

-     L1_ENDPOINT_HTTP=
-     L1_ENDPOINT_WS=

To set up an endpoint using alchemy follow these steps:

create an account
under the dropdown tab 'Apps' at the top click 'create app'
set Chain to 'Ethereum', and Network to 'Ethereum Sepolia'
Name the app 'Taiko A3 Node' or whatever you'd like
navigate back to 'Apps' dropdown at the top, choose 'Taiko A3 Node' that you just created
click 'View Key' in top right
Copy the second line "https://eth-sepolia.....' as the endpoint ENDPOINT_HTTP= input in the .env file (you may have to unhide the .env file if viewing from finder (on mac: Cntrl + Command + . )
Copy the third line 'wss://eth-sepolia....' into the ENPOINT_WS= input
save the file
close

**Comment by kshyun28** [comment]
Hello @d1onys1us, I made an initial draft for the improvements for node guide, shown in the screenshots below.

Btw, thanks for the brief guide @Eleleth77, I decided to also try using Infura.

For the initial draft of changes:
- Added brief screenshots for getting L1 endpoints on Alchemy and Infura.
- Moved the `open .env` step after getting L1 endpoints.

Would love to hear both of your feedbacks, then I can work on this if it's okay. 

<img width="667" alt="alchemy 1" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/839fa0dd-3cd3-47c9-ac2a-6cb98e49fa73">
<img width="627" alt="alchemy 2" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/dc1bfebd-0f4e-4739-b461-e4925dc6c5bd">
<img width="647" alt="alchemy 3" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/f7fa9882-39e2-4b16-8ed3-1872cacccc25">
<img width="628" alt="infura 1" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/324b6644-3770-4a54-b72d-5ab850fe8349">
<img width="634" alt="infura 2" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/d31e2d38-4314-48a7-bcd2-ebff374b19cf">

**Comment by Louisforbes147** [comment]
> ![Image](https://user-images.githubusercontent.com/13951458/245637407-51a50dee-d755-4ddf-8b3d-f035077b9d47.png)
> 
> this is too vague for the user. we can update it with something like: Make an Alchemy account -> Create new app -> Choose Sepolia -> View Key -> Place http... and wss... address in .env file -> Save

Also give out a well laid instruction and guidelines to help new users

**Comment by kshyun28** [comment]
I realized my first draft was still vague, so made another version.

# Configure Node
<img width="710" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/b9645875-5ebb-4c8c-b26e-bb83b6b87977">

## Alchemy
<img width="710" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/89203af4-711d-416a-93f1-619cfa6e449d">
<img width="811" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/925a5d2c-ff82-4cdb-9dbe-fd6237fd3ca6">
<img width="808" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/070e20c2-79b6-4fad-8357-c8207497a432">
<img width="804" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/7acbdeae-9bd0-4e00-a337-863dc32a6fb5">
<img width="846" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/b88f693e-7b16-4006-b0a2-ded01a28e3fa">

## Infura
<img width="710" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/b9645875-5ebb-4c8c-b26e-bb83b6b87977">
<img width="817" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/498b3b19-6e6d-472d-a6d6-263d4a534fa6">
<img width="816" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/5ee059c7-19c7-46d4-85cb-6d6824f9d549">
<img width="819" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/749cd042-158b-4f2a-a031-fd1fa7eac8c1">
<img width="842" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/56709653/c84ad7e8-191f-4c63-8d76-1ee87a13ee5e">

**Comment by d1onys1us** [comment]
@kshyun28 this is amazing. thank you so much for taking the time to do it. we'd be happy to accept the change if you can open a PR 👍

**Comment by kshyun28** [comment]
Hello @d1onys1us, thanks for confirming and have opened a PR. Although can't seem to add reviewers for the PR. 

It will be my first time contributing to open-source if ever. Just want to take this opportunity to ask, since am not sure if am doing things correctly.

if I were to contribute again, would the steps be:
- Open an issue or find an existing issue
- Confirm with team before working on a PR
- Open a PR for review

**Comment by d1onys1us** [comment]
> Hello @d1onys1us, thanks for confirming and have opened a PR. Although can't seem to add reviewers for the PR.
> 
> It will be my first time contributing to open-source if ever. Just want to take this opportunity to ask, since am not sure if am doing things correctly.
> 
> if I were to contribute again, would the steps be:
> 
> * Open an issue or find an existing issue
> * Confirm with team before working on a PR
> * Open a PR for review

Correct, and thanks for asking. We can improve our README to give more guidance for first time contributors like yourself about the process.
